File: mvp/src/main/java/me/giacoppo/mvp/interactor/DefaultObserver.java
package me.giacoppo.mvp.interactor;

import io.reactivex.observers.DisposableObserver;

public class DefaultObserver<T> extends DisposableObserver<T> {

    @Override
    public void onNext(T value) {

    }

    @Override
    public void onError(Throwable e) {

    }

    @Override
    public void onComplete() {

    }
}

File: mvp/src/main/java/me/giacoppo/mvp/interactor/DefaultObserver.kt
package me.giacoppo.mvp.interactor

import io.reactivex.observers.DisposableObserver

/**
 * Default observer with empty methods
 */
open class DefaultObserver<T> : DisposableObserver<T>() {

    override fun onNext(value: T) {

    }

    override fun onError(e: Throwable) {

    }

    override fun onComplete() {

    }
}
